src/settings.rs

```diff
@@ -163,34 +339,299 @@ impl WasabiSettings {
-            Ok(..) => {
-                let cfg = Self::default();
-                Self::save_to_file(&cfg);
-                cfg
-            }
-            Err(..) => Self::default(),
+    fn augment_from_args(&mut self) {
+        let matches = Command::new("wasabi")
+            .version(env!("CARGO_PKG_VERSION"))
+            .author(env!("CARGO_PKG_AUTHORS"))
+            .about(env!("CARGO_PKG_DESCRIPTION"))
+            .arg(
+                Arg::new("synth")
+                    .help("The synthesizer to use")
+                    .long_help(
+                        "The synthesizer that is used to play the MIDI. \
+                        This can either be XSynth (recommended) or KDMAPI. KDMAPI \
+                        only works if you have OmniMidi installed, and are using Windows",
+                    )
+                    .short('S')
+                    .long("synth")
+                    .value_parser(Synth::from_str),
+            )
+            .arg(
+                Arg::new("buffer-ms")
+                    .help("The amount of time events are held in the buffer")
+                    .long_help(
+                        "The amount of time that events are held in the \
+                        buffer before being played. Higher numbers may increase \
+                        performance but also increase latency.",
+                    )
+                    .short('b')
+                    .long("buffer-ms")
+                    .value_parser(f64_parser),
+            )
+            .arg(
+                Arg::new("sfz-path")
+                    .help("The path to an SFZ SoundFont")
+                    .long_help(
+                        "The path to any SFZ soundfont. In audio only mode \
+                        a soundfont must be passed either via the config file, or
+                        this command line option. In the GUI you can set this under \
+                        `Open Synth Settings > SFZ Path`",
+                    )
+                    .short('s')
+                    .long("sfz-path")
+                    .value_hint(ValueHint::FilePath),
+            )
+            .arg(
+                Arg::new("dont-limit-layers")
+                    .help("Do not apply the layer limit to the synth")
+                    .long_help(
+                        "This allows the synth to create as many layers as it \
+                        needs to play the MIDI file faithfully. Only turn this on if your \
+                        MIDI sounds bad, or your computer is running on GFuel",
+                    )
+                    .long("dont-limit-layers")
+                    .action(ArgAction::SetFalse),
+            )
+            .arg(
+                Arg::new("layer-count")
+                    .help("The amount of layers that the synthesizer can create")
+                    .long_help(
+                        "The amount of virtual pianos that the synth can create to \
```

Comment:
I think it's better to write this as "The maximum amount of voices the synth is allowed to create per key per channel." or something
Also idk about "A reasonable number for this value is the number of CPU cores your computer has."